Dockerfile

```diff
@@ -0,0 +1,134 @@
+RUN NODEJS_NAME="node-v${NODE_VERSION}-linux-x64" \
+   && NODEJS_TAR="${NODEJS_NAME}.tar.xz" \
+   && mkdir -p /home/goblin \
+   && cd /home/goblin \
+   && curl "https://nodejs.org/dist/v${NODE_VERSION}/${NODEJS_NAME}.tar.xz" -o "/home/goblin/${NODEJS_TAR}" \
+   && tar xf ${NODEJS_TAR} \
+   && mv "${NODEJS_NAME}" nodejs \
+   && rm "${NODEJS_TAR}" \
+   && corepack enable
+
+# Copy only the cargo dependency-related files first, to cache efficiently
+WORKDIR /goblin
+COPY --chown=goblin:goblin packages/backend/native-utils/Cargo.toml packages/backend/native-utils/Cargo.toml
+COPY --chown=goblin:goblin packages/backend/native-utils/Cargo.lock packages/backend/native-utils/Cargo.lock
+COPY --chown=goblin:goblin packages/backend/native-utils/src/lib.rs packages/backend/native-utils/src/
+COPY --chown=goblin:goblin packages/backend/native-utils/migration/Cargo.toml packages/backend/native-utils/migration/Cargo.toml
+COPY --chown=goblin:goblin packages/backend/native-utils/migration/src/lib.rs packages/backend/native-utils/migration/src/
+
+# Install cargo dependencies
+RUN cargo fetch --locked --manifest-path /goblin/packages/backend/native-utils/Cargo.toml
+
+
+# Copy only the dependency-related files first, to cache efficiently
+COPY --chown=goblin:goblin package.json pnpm*.yaml ./
+COPY --chown=goblin:goblin packages/backend/package.json packages/backend/package.json
+COPY --chown=goblin:goblin packages/client/package.json packages/client/package.json
+COPY --chown=goblin:goblin packages/sw/package.json packages/sw/package.json
+COPY --chown=goblin:goblin packages/firefish-js/package.json packages/firefish-js/package.json
+COPY --chown=goblin:goblin packages/megalodon/package.json packages/megalodon/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/package.json packages/backend/native-utils/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/npm/linux-x64-musl/package.json packages/backend/native-utils/npm/linux-x64-musl/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/npm/linux-arm64-musl/package.json packages/backend/native-utils/npm/linux-arm64-musl/package.json
+
+# Configure corepack and pnpm, and install dev mode dependencies for compilation
+RUN corepack install -g pnpm@8.15.4
+# && corepack use pnpm@8.12.0
+RUN pnpm i --frozen-lockfile \
+    && pnpm prune
+
+# Copy in the rest of the native-utils rust files
+COPY --chown=goblin:goblin packages/backend/native-utils packages/backend/native-utils/
+
+# Compile native-utils
+RUN pnpm run --filter native-utils build
+
+# Copy in the rest of the files to compile
+COPY --chown=goblin:goblin . ./
+RUN env NODE_ENV=production sh -c "pnpm run --filter '!native-utils' build && pnpm run gulp"
+
+# Trim down the dependencies to only those for production
+RUN pnpm i --prod --frozen-lockfile \
+    && pnpm prune
+
+
+############################# Runtime container
+FROM debian:bookworm-20240211
```

Comment:
I  should/could bump these

Reply:
Is it worth looking at alpine?

Reply:
I don't think there would be any tangible benefit, besides people wanting / being familiar with alpine. We can add it later as a separate option. The packages installed would need renaming, but conceptually it's the same. 

Debian base image is 117MB, alpine is 7MB, but the Goblin build is pretty heavy, so the size gap would be a lot smaller in reality - plus there's 2GB of npm packages, so even hypothetical 100MB size difference would not change anything in a 2.35GB final image - plus the layers are going to be cached anyway.

It might make sense later if we somehow split frontend, backend and the service worker or other stuff that can run independently, but that sounds like a lot of work.